clang-tools-extra/clang-doc/JSONGenerator.cpp

```diff
@@ -234,10 +267,12 @@ static void serializeArray(const Container &Records, Object &Obj,
   json::Value RecordsArray = Array();
   auto &RecordsArrayRef = *RecordsArray.getAsArray();
   RecordsArrayRef.reserve(Records.size());
-  for (const auto &Item : Records) {
+  for (size_t Index = 0; Index < Records.size(); ++Index) {
```

Comment:
I don't think `enumerate()` will make this nicer, but its available if you do. It's not the most widely used pattern, but I'd hate for you not to know its around (like I did for a couple years).

Reply:
Yeah I saw that Peter had used it previously for the same function but I really didn't see the benefit over this if I couldn't use a range-based loop naturally. I think the old enumeration would actually serialize `"End" = false` everywhere else.